packages/Kokkos/src/details/DTK_DetailsTreeTraversal.hpp

```diff
@@ -181,28 +126,19 @@ KOKKOS_FUNCTION void nearest_query( BVH<NO> const bvh, Point const &query_point,
             {
                 double child_distance =
                     distance( query_point, child->bounding_box );
-                queue.emplace( child, child_distance );
+                queue.push( child, child_distance );
             }
         }
     }
-
-    n_indices = candidate_list.size();
-    unsigned int pos = 0;
-    for ( auto const &elem : candidate_list._sorted_list )
-    {
-        indices[pos] = elem.first;
-        ++pos;
-    }
-#endif
 }
 
 template <typename NO, typename Predicate>
 int query_dispatch( BVH<NO> const bvh, Predicate const &pred,
                     Kokkos::View<int *, typename NO::device_type> out,
                     NearestPredicateTag )
 {
-    unsigned int constexpr n_max_indices = 1000;
-    int indices[n_max_indices];
+    unsigned int constexpr max_n_indices = 1000;
```

Comment:
I think this would be good to move all the `constexpr` variables for the size in a separate file so that we don't need to dig to change the value.